### ISSUE TITLE
Search syntax: '+' should mean 'any category' 

### ISSUE THREAD

**Issue by heikkil** [skeeto/elfeed#67]
I my opinion the plus character '+' should match 'any category' in a search string.

I keep trying to reduce key strokes that I repeat a lot. I usually start elfeed session by reading the most common new categories first and for me that is 'Daily News' : '+n'. When I am done with those, I replace that part of the search string with '+j' that lists new science journal feeds. After that I go for the next most numerous category. (They all have one letter codes) Backspacing the search string from '... +n' to '...+' still does not display any articles because '+' has a special meaning to note a category and does not match anything. 

However, in practice all articles have some category (default being something like  '+read' or but there is no reason to ever show that as a string). Therefore a better semantic  for '+' is 'any category'.

In other words, the effect of the following two search strings should be the same:
```` @1-month-ago +unread +```` 
```` @1-month-ago +unread ```` 

That would allow me to use the former to see the most numerous category, and type in the corresponding one letter category and enter to start reading those articles. ... And it makes more semantic sense, at least to me.

**Comment by skeeto** [comment]
(In my response, for consistency I'll be using the standard Elfeed terms
for these things: s/articles/entries/, s/categories/tags/.)

I like this train of thought. Right now a lone + will find all entries
tagged with the empty symbol ('##). Perhaps someone out there is taking
advantage of this, but I think that's a pretty useless tag, so I'm fine
with taking away the ability to use it.

As a side note: for quickly switching between different common filters,
some people bind keys (elfeed-search-mode-map) to change the filter in
effect without needing to manually edit the filter. This would allow you
to use full tag names, rather than single characters, without having to
type them out.
